File: 2825-Make-String-a-Subsequence-Using-Cyclic-Increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        
        int i=0,j=0;
        for (; i<str1.size() ;i++){
            if (str1[i]=='z'){
                if (str2[j]=='z' || str2[j]=='a') {
                    j++;
                    if (j==str2.size()) return true;
                }
            }else{
                if (str1[i]==str2[j] || str1[i]+1==str2[j]){
                    j++;
                    if (j==str2.size()) return true;
                }
            }
        }
        return false;
        
    }
};

File: 2825-Make-String-a-Subsequence-Using-Cyclic-Increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        j=0
        for i in range(len(str1)):
            if str1[i]==str2[j]:
                j+=1
                if j==len(str2) : return True
            else:
                if str1[i]=='z':
                    if str2[j]=='a':
                        j+=1
                        if j==len(str2): return True
                else:
                    if str2[j]==chr(ord(str1[i])+1):
                        j+=1
                        if j==len(str2): return True
        return False
                        